### PR TITLE
ODIS CONTRIBUTING.md / dev guide 

### PR DESCRIPTION
**Pull request by alecps**
### Description

- Adds a CONTRIBUTING.md file with instructions on how to set up your dev environment and navigate the project.
- Moves `./kubernetes-deployment` directory out of `./docs` and up one level as it is not external facing documentation for SocialConnect users

### Backwards compatibility

Yes

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -9,8 +9,92 @@ If you wish to submit more complex changes, please sync with a core developer fi
...
+```
+
+#### Install node modules and build with yarn
+
+We use yarn berry. You may need to run `corepack enable`. This doesn't setup yarn itself but is a node.js feature for enabling package managers.
+
+If you don't have yarn installed, run
+
+```bash
+npm i -g yarn
```
Comment (aaronmgdr) [inline .github/CONTRIBUTING.md:63]: actually im fairly certain that you dont need to do this.

Comment (aaronmgdr) [reply .github/CONTRIBUTING.md:63]: as it would be installing yarn 1

Comment (alecps) [reply .github/CONTRIBUTING.md:63]: Are you suggesting I remove this step entirely or edit it. If the latter, could you please specify the command you would like here?

Comment (alecps) [reply .github/CONTRIBUTING.md:63]: I guess I don't see how yarn could be installed automatically if it hasn't been previously. I also would think that the latest version would be installed by this command. Could you fill me in?

**Review by soloseng — APPROVED** [review]

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (aaronmgdr) [reply .github/CONTRIBUTING.md:63]: yeah so i was a bit confused about this at first too. but `corepack enable` gives the `yarn` command and then it seems full yarn version that is needed is installed when the project is being installed 

https://yarnpkg.com/getting-started/install

only version 1 of yarn is on npm

Comment (alecps) [reply .github/CONTRIBUTING.md:63]: got it, thanks!

**Pull request merged**